include/seqan3/core/simd/view_to_simd.hpp

```diff
@@ -0,0 +1,719 @@
+
+    /*!\name Auxiliary types
+     * \{
+     */
+    using inner_range_type = value_type_t<urng_t>; //!< The inner range type.
+    using chunk_type = std::array<simd_t, simd_traits<simd_t>::length>; //!< The underlying type to hold the chunks.
+    using scalar_type = typename simd_traits<simd_t>::scalar_type; //!< The scalar type.
+    //!\brief The SIMD type with maximal number of lanes for the current arch.
+    using max_simd_type = simd_type_t<uint8_t, simd_traits<simd_t>::max_length>;
+    //!\}
+
+    /*!\name Auxiliary variables
+     * \{
+     */
+    //!\brief Check if fast load is enabled.
+    static constexpr bool fast_load = std::ranges::ContiguousRange<inner_range_type> &&
+                                      std::SizedSentinel<std::ranges::iterator_t<inner_range_type>,
+                                                         std::ranges::sentinel_t<inner_range_type>> &&
+                                      sizeof(alphabet_rank_t<value_type_t<inner_range_type>>) == 1 &&
+                                      sizeof(scalar_type) <= 2; // micro benchmark suggest using int8_t or int16_t has best performance.
+
+    //!\brief The size of one chunk. Equals the number of elements in the vector.
+    static constexpr uint8_t chunk_size = simd_traits<simd_t>::length;
+    //!\brief The number of chunks that can be gathered with a single load.
+    static constexpr uint8_t chunks_per_load = simd_traits<simd_t>::max_length / chunk_size;
+    //!\brief The total number of chunks that can be cached.
+    static constexpr uint8_t total_chunks = fast_load ? (chunks_per_load * chunks_per_load) : 1;
+    //!\brief The alphabet size.
+    static constexpr auto alphabet_size = alphabet_size<value_type_t<inner_range_type>>;
+    // NOTE: only needed for doxygen, otherwise the generated constructor signature that takes the expression
+    // above as default value will be destroyed :(
+    //!\}
+
+    // Forward declare class' iterator type. See definition below.
+    struct iterator_type;
+
+public:
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    constexpr view_to_simd() = default; //!< Defaulted.
+    constexpr view_to_simd(view_to_simd const &) = default; //!< Defaulted.
+    constexpr view_to_simd(view_to_simd &&) = default; //!< Defaulted.
+    constexpr view_to_simd & operator=(view_to_simd const &) = default; //!< Defaulted.
+    constexpr view_to_simd & operator=(view_to_simd &&) = default; //!< Defaulted.
+    ~view_to_simd() = default; //!< Defaulted.
+
+    /*!\brief Construction from the underlying range.
+     * \param[in] urng The underlying range.
+     * \param[in] padding_value The value used to fill up smaller sequences.
+     */
+    constexpr view_to_simd(urng_t urng, scalar_type const padding_value = alphabet_size) :
+        urng{std::move(urng)},
+        cached_padding_vector{simd::fill<simd_t>(padding_value)},
```

Comment:
why is the thing called cached? and not just padding_simd?

Reply:
I will rename it